contracts/FeralfileEnglishAuction.sol

```diff
@@ -0,0 +1,378 @@
+        for (uint256 i = 0; i < auctions_.length; i++) {
+            EnglishAuction memory auc_ = auctions[auctions_[i].id];
+
+            require(
+                auc_.id == 0,
+                "FeralfileEnglishAuction: auction already exist"
+            );
+
+            require(
+                auctions_[i].startAt >= block.timestamp,
+                "FeralfileEnglishAuction: auction start time should be in the future"
+            );
+
+            require(
+                auctions_[i].endAt > auctions_[i].startAt,
+                "FeralfileEnglishAuction: auction end time should be after start time"
+            );
+
+            require(
+                auctions_[i].extendDuration >= 0,
+                "FeralfileEnglishAuction: auction extend duration should be greater than or equal 0"
+            );
+
+            require(
+                auctions_[i].extendThreshold >= 0,
+                "FeralfileEnglishAuction: auction extend threshold should be greater than or equal 0"
+            );
+
+            require(
+                auctions_[i].minIncreaseFactor > 0,
+                "FeralfileEnglishAuction: auction min increase factor should be greater than 0"
+            );
+
+            require(
+                auctions_[i].minIncreaseAmount > 0,
+                "FeralfileEnglishAuction: auction min increase amount should be greater than 0"
+            );
+
+            require(
+                auctions_[i].minPrice > 0,
+                "FeralfileEnglishAuction: auction min price should be greater than 0"
+            );
+
+            auctions[auctions_[i].id] = auctions_[i];
+        }
+    }
+
+    function listAuctionLatestStatus(
+        bytes32[] memory aucIds_
+    ) external view returns (LatestAuctionStatus[] memory) {
+        LatestAuctionStatus[] memory results = new LatestAuctionStatus[](
+            aucIds_.length
+        );
+        for (uint i = 0; i < aucIds_.length; i++) {
+            // get the auction object
+            EnglishAuction memory auction_ = auctions[aucIds_[i]];
+            // get the last auction bid object
+            results[i].latestBid = auctionLatestBid[aucIds_[i]];
+            results[i].endAt = auction_.endAt;
+            results[i].isSettled = auction_.isSettled;
+        }
+        return results;
+    }
+
+    function isValidBidRequest(
+        bytes32 aucId_,
+        address bidder_,
+        uint256 amount_
+    ) external view returns (bool) {
+        // get the auction object
+        EnglishAuction memory auction_ = auctions[aucId_];
+        // get the last auction bid object
+        Bid memory lastBid_ = auctionLatestBid[aucId_];
+        // verify the bid is valid
+        return
+            _isValidBidRequest(
+                auction_,
+                lastBid_,
+                bidder_,
+                amount_,
+                block.timestamp
+            );
+    }
+
+    function _isValidBidRequest(
```

Comment:
I feel like this function is inconsistent. It returns a `bool` means that it validate the new bid if it's correct but somehow, we also throw error when validating the bid. I'd suggest either throw error (dont return `bool`) or return `bool` without any errors.